test/benchgc/src/benchgc/__main__.py

```diff
@@ -0,0 +1,355 @@
+################################################################################
+# Copyright 2024 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+
+import sys
+import argparse
+
+import torch
+
+from benchgc.arg.arg import Arg
+from benchgc.arg import (
+    fill_tensor,
+    compare_tensor,
+    set_default_compare,
+    set_default_fill,
+)
+from typing import Dict, List
+import runner
+import benchgc.util
+import benchgc.mlir.util
+import gc_mlir.ir
+from gc_mlir.graph_compiler import GraphCompiler
+import os
+from benchgc.mlir.arg import get_mlir_args
+from benchgc.pattern.mlp import MLP
+from benchgc.mlir.bench import py_timeit_bench, mlir_wrapper_bench
+import json
+
+def performance_testing(flags, module, args):
+    gc_args: List[torch.Tensor | int] = []
+    gc_tensors: Dict[str, torch.Tensor] = {}  
+    for i in range(len(args)):
+        tensor = fill_tensor(flags, args[i], i)
+        gc_tensors["%arg" + str(i)] = tensor
+        if args[i].scalar:
+            gc_args.append(tensor.data_ptr())
+        else:
+            gc_args.append(tensor)
```

Comment:
Please share the tensor filling code with correctness mode.
It is a better to handle this part in the main workflow.